src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs

```diff
@@ -329,6 +330,108 @@ public bool IsNodeAddedRecently
             }
         }
 
+        /// <summary>
+        /// Gets a value indicating whether this model is frozen.
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if this instance is frozen; otherwise, <c>false</c>.
+        /// </value>
+        public bool IsFrozen
+        {
+            get
+            {              
+                return NodeModel.IsFrozen;
+            }
+            set
+            {
+                NodeModel.IsFrozen = value;
+            }
+        }
+        
+        /// <summary>
+        /// Gets a value indicating whether Freeze property on the node is checked
+        /// </summary>
+        /// <value>
+        ///   <c>true</c> if node is not frozen and CanExecute; otherwise, <c>false</c>.
+        /// </value>        
+        public bool NodeRunChecked
```

Comment:
I find `NodeRunChecked` and `NodeRunEnabled` a bit confusing. The problem is that you need to manifest the "frozen" state of the node in several ways in the UI:
  1. Node is not frozen - Should be unchecked and enabled.
  2. Node is frozen and is the root of the freeze operation - Should be checked and enabled.
  3. Node is frozen and is not the root - Should be checked and disabled.